packages/experimental/TabList/src/TabList/useTabList.ts

```diff
@@ -36,11 +36,29 @@ export const useTabList = (props: TabListProps): TabListInfo => {
   } = props;
 
   const data = useSelectedKey(selectedKey || defaultSelectedKey || null, onTabSelect);
+  const selectedTabKey = selectedKey ?? data.selectedKey;
 
-  // selectedTabRef should be set to default tabbable element.
-  const [selectedTabRef, setSelectedTabRef] = React.useState(React.useRef<View>(null));
+  // focusedTabRef should be set to default tabbable element.
+  const [focusedTabRef, setFocusedTabRef] = React.useState(React.useRef<View>(null));
   const [invoked, setInvoked] = React.useState(false);
   const [tabKeys, setTabKeys] = React.useState<string[]>([]);
+  const [allTabsDisabled, setAllTabsDisabled] = React.useState(false);
+
+  // These
+  const tabRefMap = React.useRef<{ [key: string]: React.RefObject<View> }>({}).current;
+  const disabledStateMap = React.useRef<{ [key: string]: boolean }>({}).current;
+
+  const updateTabRef = (key: string, ref: React.RefObject<View>) => (tabRefMap[key] = ref);
```

Comment:
>(key: string, ref: React.RefObject<View>) => (tabRefMap[key] = ref); [](http://example.com/codeflow?start=23&length=68)

This should be wrapped in a useCallback, otherwise the function is recreated on every render and the function "pointer" will become outdated in the tabs